src/contexts/RDKitContext.tsx

```diff
@@ -10,26 +11,34 @@ export interface RDKitContextValue {
 export type RDKitProviderProps = PropsWithChildren<{
   cache?: RDKitProviderCacheOptions;
   preferCoordgen?: boolean;
+  rdkitWorkerPath?: string;
+  initialRdkitInstance?: RDKitModule;
 }>;
 
 // force default context to be undefined, to check if package users have wrapped it with the required provider
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const RDKitContext = React.createContext<RDKitContextValue>(undefined as any);
 
-export const RDKitProvider: React.FC<RDKitProviderProps> = ({ cache = {}, preferCoordgen = false, children }) => {
+export const RDKitProvider: React.FC<RDKitProviderProps> = ({
+  cache = {},
+  preferCoordgen = false,
+  rdkitWorkerPath,
+  initialRdkitInstance,
+  children,
+}) => {
   const [worker, setWorker] = useState<Worker | null>(null);
 
   useEffect(() => {
     let isProviderMounted = true;
     let workerInstance: Worker | null = null;
 
     const initialise = async () => {
-      workerInstance = initWorker();
+      workerInstance = initWorker(rdkitWorkerPath);
       // await rdkit module init in worker before starting using the worker
       await postWorkerJob(workerInstance, {
         actionType: RDKIT_WORKER_ACTIONS.INIT_RDKIT_MODULE,
         key: 'worker-init',
-        payload: { cache, preferCoordgen },
+        payload: { cache, preferCoordgen, initialRdkitInstance },
```

Comment:
I thought I couldn't pass an `rdkitModule` from main thread to the worker thread, as it failed to serialize it, did this work for you? passing around `rdkitModule` from main to worker
@atfera

Reply:
You are right, cant be cloned. I removed it.